File: guides/micronaut-jaxrs-jdbc/java/src/main/java/example/micronaut/PetSave.java
package example.micronaut;

import io.micronaut.core.annotation.NonNull;
import io.micronaut.serde.annotation.Serdeable;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;

@Serdeable
public class PetSave {
    @NonNull
    @NotBlank
    private final String name;

    @NonNull
    @NotNull
    private final PetType type;

    public PetSave(@NonNull String name, @NonNull PetType type) {
        this.name = name;
        this.type = type;
    }

    @NonNull
    public String getName() {
        return name;
    }

    @NonNull
    public PetType getType() {
        return type;
    }
}

File: guides/micronaut-jaxrs-jdbc/python/src/example/micronaut/pet_save.py
from dataclasses import dataclass
from typing import Annotated

from jakarta.validation.constraints import NotBlank, NotNull
from micronaut.serde.annotation import Serdeable

from .pet_type import PetType


@dataclass
@Serdeable
class PetSave:
    name: Annotated[str, NotBlank]
    type: Annotated[PetType, NotNull]
